src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -426,6 +431,67 @@ def create(self, daemon_spec: CephadmDaemonSpec) -> str:
+            result = subprocess.run(['radosgw-admin', '--key=%s'%keyring, '--user', 'rgw.%s'%rgw_id, 'realm', 'create', '--rgw-realm=%s'%spec.rgw_realm, '--default'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            self.mgr.log.info('created realm: %s'%spec.rgw_realm)
+        else:
+            try:
+                j = json.loads(result.stdout)
+                if 'realms' not in j or spec.rgw_realm not in j['realms']:
+                    result = subprocess.run(['radosgw-admin', '--key=%s'%keyring, '--user', 'rgw.%s'%rgw_id, 'realm', 'create', '--rgw-realm=%s'%spec.rgw_realm, '--default'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                    self.mgr.log.info('created realm: %s'%spec.rgw_realm)
+            except Exception as e:
+                raise OrchestratorError('failed to parse realm info')
+
+        # get zonegroup
+        result = subprocess.run(['radosgw-admin', '--key=%s'%keyring, '--user', 'rgw.%s'%rgw_id, 'zonegroup', 'list', '--format=json'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        #create zonegroup if needed
+        if not result.stdout:
+            result = subprocess.run(['radosgw-admin', '--key=%s'%keyring, '--user', 'rgw.%s'%rgw_id, 'zonegroup', 'create', '--rgw-zonegroup=default', '--master', '--default'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            self.mgr.log.info('created zonegroup: default')
+        else:
+            try:
+                j = json.loads(result.stdout)
+                if 'zonegroups' not in j or 'default' not in j['zonegroups']:
+                    result = subprocess.run(['radosgw-admin', '--key=%s'%keyring, '--user', 'rgw.%s'%rgw_id, 'zonegroup', 'create', '--rgw-zonegroup=default', '--master', '--default'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                    self.mgr.log.info('created zonegroup: default')
+            except Exception as e:
+                raise OrchestratorError('failed to parse zonegroup info')
+
+        #get zones
+        result = subprocess.run(['radosgw-admin', '--key=%s'%keyring, '--user', 'rgw.%s'%rgw_id, 'zone', 'list', '--format=json'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        #create zone if needed
+        if not result.stdout:
+            result = subprocess.run(['radosgw-admin', '--key=%s'%keyring, '--user', 'rgw.%s'%rgw_id, 'zone', 'create', '--rgw-zonegroup=default', '--rgw-zone=%s'%spec.rgw_zone, '--master', '--default'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            self.mgr.log.info('created zone: %s'%spec.rgw_zone)
+        else:
+            try:
+                j = json.loads(result.stdout)
+                if 'zones' not in j or spec.rgw_zone not in j['zones']:
+                    result = subprocess.run(['radosgw-admin', '--key=%s'%keyring, '--user', 'rgw.%s'%rgw_id, 'zone', 'create', '--rgw-zonegroup=default', '--rgw-zone=%s'%spec.rgw_zone, '--master', '--default'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
```suggestion
                    cmd = ['radosgw-admin',
                           '--key=%s '% keyring,
                           '--user', 'rgw. %s' % rgw_id,
                           'zone', 'create', 
                           '--rgw-zonegroup=default',
                           '--rgw-zone=%s' % spec.rgw_zone,
                           '--master', '--default']
                    result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)

```

nit .. these lines are really long .. maybe break them up to increase readability?